### ISSUE TITLE
keyboard controls broken in single selection mode

### ISSUE THREAD

**Issue by severinraez** [isteven/angular-multi-select#201]
Hi

This component (v 3.0.0) raises an error if initialized as single select (see http://isteven.github.io/angular-multi-select/#/demo-single-selection-mode)

As I only need single selection, I hacked something together that works for me. I thought a pointer would be helpful to you. Changing the line 

    var checkboxes = element.children().children().next().children().next()[ 0 ].getElementsByTagName( 'input' );

to

    var checkboxes = element.children().children().next().children()[0].getElementsByTagName('input');

resolves the issue.

**Comment by isteven** [comment]
Hi @severinraez , 

Thanks for letting me know. 

There's an underlying bug in this. Basically, it couldn't find some elements because of different DOM structure of the directive, depending on how we configure the multi select. 

I'm working on a fix now.

**Comment by cp2587** [comment]
+1.
Nice widget otherwise

**Comment by boje** [comment]
I am experience issue that seems to be related to this. If 0 or 1 disable selection in dropdown.
If I remove `.next()` then the problem seem to be solve.

`var inputField = element.children().children().next().children().children()[0].getElementsByTagName('input');`

`var clearButton = element.children().children().next().children().children().next()[0].getElementsByTagName('button');`

**Comment by jmls** [comment]
ah, I just came across this as well

**Comment by fortesl** [comment]
Happens in multi-select mode as well...

**Comment by forrestab** [comment]
Instead of blanking the helper-elements property, I ended up adding the following css.

```css
.multiSelect .checkboxLayer .helperContainer {
    display: none;
}
```

**Comment by isteven** [comment]
Fixed in v4.0.0
